app/src/main/java/com/google/droidjump/StartFragment.java

```diff
@@ -50,19 +58,23 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     }
 
     private void play(View view) {
-        findNavController(view).navigate(R.id.action_start_screen_to_game_screen);
+        arguments.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, activity.getCurrentLevel());
+        findNavController(view).navigate(R.id.action_start_screen_to_game_screen, arguments);
     }
 
     private void chooseLevel(View view) {
         findNavController(view).navigate(R.id.action_start_screen_to_levels_screen);
     }
 
     private void startNewGame(View view) {
-        findNavController(view).navigate(R.id.action_start_screen_to_game_screen);
+        // Removing data from SharedPreferences
+        activity.resetGameData();
+        // Passing data to arguments
+        arguments.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, 1);
```

Comment:
Does not resetGameData already set GAME_VIEW_CURRENT_LEVEL_STRING to 1?

Reply:
gameData - SharedPreferences storage, arguments - local bundle with arguments that only are need in this view.
When you run activity.resetGameData() it updates gameData but not local arguments that we are passing to another view.

Reply:
(We don't need to store and pass unnecessary data in this case)